### PR TITLE
Fix formatOnSave bug with eslintignore

### LINKED ISSUE #347: "Ignore Files in `.eslintignore`" does not work
By default this is enabled.
I added "test" in .eslintignore, so that linting ignores test files. Seems to work. Also disabled prettier formatting for my test files.
I disabled this feature in the prettier-atom config so I can prettify the test files.
I found that it did no prettifying on any of my test files on save.
I restarted atom and found the same.
Lastly, I removed "test" from .eslintignore and, on save of my test file, the prettifying started working again. So it definitely seems like this feature is having no effect.

I've done some research - I'm certain this is a feature specific to prettier-atom, not something in prettier or eslint themselves.

I'm afraid I can't figure out how to run Prettier: Debug (sorry!). But I can tell you I'm on 0.46.0 of the plugin.

### PR DESCRIPTION
**Pull request by olsonpm**
- check "shouldRespectEslintignore" setting prior to checking whether
  the filepath should be eslintignore'd
- add tests for
  : should respect and is eslintignored
  : should not respect and is eslintignored
  : is not eslintignored

fixes #347

### CODE REVIEW

File: src/formatOnSave/shouldFormatOnSave.js [hunk 1]
```diff
@@ -55,7 +57,7 @@ const shouldFormatOnSave: (editor: TextEditor) => boolean = _.overEvery([
     isFilePathWhitelisted,
     _.overEvery([noWhitelistGlobsPresent, filePathDoesNotMatchBlacklistGlobs]),
   ]),
-  isFilePathNotEslintignored,
+  shouldNotEslintIgnore,
```
Comment (robwise) [inline src/formatOnSave/shouldFormatOnSave.js]: This name doesn't seem right to me? We still format even when we should EslintIgnore (as long as the file doesn't match the eslintignore glob). But this name makes it sound like we are only checking the config option.

What about just inlining the `shouldNotEslintIgnore` function? We do this for the disabling checks below.

Comment (olsonpm) [reply src/formatOnSave/shouldFormatOnSave.js]: I see what you mean.  My intent was to say that we should not ignore it due to eslint.

Comment (robwise) [reply src/formatOnSave/shouldFormatOnSave.js]: Hmm, it seems like the good name for it is already taken: `isFilePathEslintIgnored`. Perhaps we should rename the existing one to something like `doesFilePathMatchEslintIgnore` and then we can use that name?

**Comment by robwise** [comment]
@olsonpm shoot, sorry I didn't catch this before, but do you think you could squash down and use the conventional changelog standard for the commit message? Something like `fix(eslintignore): respect eslintignore setting`. Otherwise, the changelog (which is auto-generated from commit messages) won't pick up the commit.

**Comment by olsonpm** [comment]
ah sorry about that.  Thanks for letting me know

**Comment by robwise** [comment]
Thanks for the fix!

**Comment by olsonpm** [comment]
one of these days I'll stop skimming what people write and actually read it heh

**Pull request merged**